vhdbuilder/publish/sig-version-create.sh

```diff
@@ -6,14 +6,23 @@
 [[ -z "${RG_NAME}" ]] && echo "RG_NAME is not set" && exit 1
 [[ -z "${GALLERY_NAME}" ]] && echo "GALLERY_NAME is not set" && exit 1
 [[ -z "${IMAGEDEFINITION_NAME}" ]] && echo "IMAGEDEFINITION_NAME is not set" && exit 1
-[[ -z "${IMAGE_VERSION}" ]] && echo "IMAGE_VERSION is not set" && exit 1
 #TARGET_REGIONS must be set in the following format region=replicacount "westus2=1 eastus=4 uksouth=3"
 [[ -z "${TARGET_CREATE_REGIONS}" ]] && echo "TARGET_CREATE_REGIONS is not set" && exit 1
 [[ -z "${MANAGED_IMAGE_RG_NAME}" ]] && echo "MANAGED_IMAGE_RG_NAME is not set" && exit 1
 [[ -z "${VHD_SOURCE}" ]] && echo "VHD_SOURCE is not set" && exit 1
 [[ -z "${OS_NAME}" ]] && echo "OS_NAME is not set" && exit 1
+# Windows image version use a different pattern than Linux. Linux image version depends on user specified one, 
+# which is typicall current date, While Windows depends on OS version and current date to track the origin of the image
+[[ "Windows" != ${OS_NAME} ]] && [[ -z "${IMAGE_VERSION}" ]] && echo "IMAGE_VERSION is not set for Linux" && exit 1
+[[ "Windows" = ${OS_NAME} ]] && [[ -z "${WINDOWS_VERSION}" ]] && echo "WINDOWS_VERSION is not set" && exit 1
 [[ -z "${HYPERV_GENERATION}" ]] && echo "HYPERV_GENERATION is not set" && exit 1
 
+if [ "Windows" = ${OS_NAME} ]; then
+  # Windows image version in SIG follows the pattern: ${WINDOWS_VERSION}.$(date +"%y%m%d"), e.g. 17763.1457.200909
+  version_date=$(date +"%y%m%d")
```

Comment:
When publishing a Windows VHD as both PIR and SIG, it should be better to use the same version.

Reply:
If you publish the VHD in RIP and SIG in the same day, it will be the same version.

Reply:
How can we assure this? For example, there is an issue in pipeline and need to republish it after fixing the issue.

Reply:
If we have something wrong in Marketplace and we have to change the VHD version, should we also change the image in SIG and rp code correspondingly?

Reply:
We use the version in Marketplace as the base since PIR takes longer time. For the scenario you mentioned, we should publish a new SIG. But if PIR is ready but SIG is not, we should not republish a new PIR.
Another corner case is that two jobs are almost done in ~24:00

Reply:
Then I would prefer we don't use os version in SIG. more simple.

Reply:
What do you mean "don't use os version in SIG"?
To use the same image version in PIR and SIG, the benefit is known. We have a test case for Linux to check this and we will add one for Windows later.

Reply:
Please do it, thanks.

Reply:
@AbelHu  is right. we want SIG and PIR images to on the same version for node image upgrade feature.

Reply:
Use image version suffix obtained from PIR, and updated [doc](https://msazure.visualstudio.com/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/84724/windows-VHD-build-and-release) and the image version filed of [release pipeline](https://msazure.visualstudio.com/CloudNativeCompute/_releaseDefinition?definitionId=397&_a=definition-variables)